opm/input/eclipse/Python/PyRunModule.cpp

```diff
@@ -82,10 +67,84 @@ py::cpp_function py_actionx_callback(const std::function<void(const std::string&
+            return result.cast<bool>();
+        } catch (const std::exception& e2) {
+            std::cerr << "Exception thrown trying to call run(ecl_state, schedule, report_step, summary_state, actionx_callback) function of " << this->module_name << ": " << e.what() << std::endl;
+            std::cerr << "Tried calling run(ecl_state, schedule, report_step, summary_state), exception thrown: " << e2.what() << std::endl;
+            throw e2;
+        }
+    }
+}
+
+bool PyRunModule::run(EclipseState& ecl_state, Schedule& sched, std::size_t report_step, SummaryState& st, const std::function<void(const std::string&, const std::vector<std::string>&)>& actionx_callback) {
+    // The attributes need to be set before the user defined module is loaded; it needs to be set in every call.
+    this->opm_embedded.attr("current_report_step") = report_step;
+
+    if (!this->module) {
+        // The attributes need to be set before the user defined module is loaded; they need to be set only once since they are shared pointers.
+        this->opm_embedded.attr("current_schedule") = &sched;
+        this->opm_embedded.attr("current_summary_state") = &st;
+        this->opm_embedded.attr("current_ecl_state") = &ecl_state;
+        try {
+            if (!this->module_path.empty()) {
+                py::module sys = py::module::import("sys");
+                py::list sys_path = sys.attr("path");
+                {
+                    bool have_path = false;
+                    for (const auto& elm : sys_path) {
+                        const std::string& path_elm = static_cast<py::str>(elm);
+                        if (path_elm == this->module_path)
+                            have_path = true;
+                    }
+                    if (!have_path)
+                        sys_path.append(py::str(this->module_path));
+                }
+            }
+            this->module = py::module::import(this->module_name.c_str());
+        } catch (const std::exception& e) {
+            std::cerr << "Exception thrown when loading Python module " << this->module_name << ": " << e.what() << std::endl;
+            throw e;
+        } catch (...) {
+            std::cerr << "General exception thrown when loading Python module " << this->module_name << "!" << std::endl;
+        }
+
+        if (this->module.is_none()) {
+            throw std::runtime_error("Syntax error when loading Python module: " + this->module_name);
+        }
+
+        this->module.attr("storage") = this->storage;
+
+        if (py::hasattr(this->module, "run")) {
+            std::cout << R"(PYACTION can be used without a run(ecl_state, schedule, report_step, summary_state, actionx_callback) function, its arguments are available as attributes of the module opm_embedded, try the following in your python script:
```

Comment:
I think we should use `OpmLog::info` here. Then the message is also in the output files.